Lib/test/test_pydoc/test_pydoc.py

```diff
@@ -76,6 +76,11 @@ class A(builtins.object)
      |  __weakref__%s
 
     class B(builtins.object)
+     |  Methods defined here:
+     |
+     |  __annotate__ = <annotations of B>(...)
+     |
+     |  ----------------------------------------------------------------------
```

Comment:
What do you suggest? Should we special case `__annotate__` in pydoc somewhere?

Reply:
This is fundamentally caused by the fact that `__annotate__` has its `__name__` set to a string with spaces in it, right? Why does `__annotate__` not have its `__name__` set to the name it can be accessed by (`__annotate__`), like nearly every other method that you can access from a class?

```pycon
>>> class Foo:
...     x: int
...     def y(self): pass
...     
>>> print(Foo.y, Foo.y.__name__, Foo.__annotate__, Foo.__annotate__.__name__, sep="\n")
<function Foo.y at 0x104e99430>
y
<function Foo.<annotations of Foo> at 0x104e992d0>
<annotations of Foo>
```

Reply:
I guess that means every `__annotate__` function would have the same name when it shows up in stack traces, which maybe isn't the worst?

It would mean that in a stack trace, instead of this:

```
% ./python.exe -c '''
def f(x: y): pass

print(f.__annotations__)
'''
Traceback (most recent call last):
  File "<string>", line 4, in <module>
    print(f.__annotations__)
          ^^^^^^^^^^^^^^^^^
  File "<string>", line 2, in <annotations of f>
    def f(x: y): pass
             ^
NameError: name 'y' is not defined
```

You'd see "`line 2, in __annotate__`".

Reply:
> I guess that means every `__annotate__` function would have the same name when it shows up in stack traces, which maybe isn't the worst?

I think that's probably not the worst. For most method calls, you only generally get the name of the method in the stack trace, you don't generally get the qualname:

```pycon
>>> class Foo:
...     def __init__(self):
...         causes_namerror
...         
>>> Foo()
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    Foo()
    ~~~^^
  File "<python-input-0>", line 3, in __init__
    causes_namerror
NameError: name 'causes_namerror' is not defined
```

I _like_ the `<annotations of f>` thing showing up in the stack trace -- since this is a function that has no source code, that has been magically inserted by the compiler, I do agree that it's helpful to display it like that. But IMHO, I think it's more important to maintain the invariant here that for a method in a class namespace, `__name__` is almost always a valid identifier, and that `__name__` is almost always the identifier by which you can access that method using the `.` operator on the class.

If we want to have tracebacks show `<annotations of f>` rather than `__annotate__`, maybe we could add some special-casing to `traceback.py`? But again, I don't think it's the worst thing in the world if the traceback just shows `__annotate__`.

Reply:
Alex's argument makes sense to me, and I agree with it as far as tracebacks go. The line number should be sufficient to clarify which `__annotate__`, and the frame we're talking about here is one that is just totally absent in the status quo, so we aren't making things worse.

But I think renaming to just `__annotate__` will lead to confusing symbol tables, at the very least, whenever multiple functions are defined in one scope. Since function `__annotate__` scopes are sibling to the function scope, if you have multiple functions defined in a scope you currently get this:

```
>>> code = """
... def f() -> None: ...
... def g() -> int: ...
... """
>>> st = symtable.symtable(code, "test", "exec")
>>> st
<SymbolTable for module test>
>>> st.get_children()
[<SymbolTable for <annotations of f> in test>, <Function SymbolTable for f in test>, <SymbolTable for <annotations of g> in test>, <Function SymbolTable for g in test>]
```

With the proposed change here, that last line would show two different `__annotate__` symbol tables, with no way (other than implicitly by ordering, I guess) to differentiate which one goes with which function.

I'm not sure in practice how much of a problem this would be.

Reply:
I pushed a change renaming the function to `__annotate__`.